src/test/TestUtils.cpp

```diff
@@ -3,54 +3,25 @@
 // of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
 
 #include "TestUtils.h"
-
-#include "herder/HerderImpl.h"
-#include "herder/LedgerCloseData.h"
 #include "util/make_unique.h"
 
 namespace stellar
 {
 
-LedgerManagerEditableVersion::LedgerManagerEditableVersion(Application& app,
-                                                           uint32_t baseFee)
-    : LedgerManagerImpl{app}, mBaseFee{baseFee}
-{
-}
-
-uint32_t
-LedgerManagerEditableVersion::getCurrentLedgerVersion() const
-{
-    return mCurrentLedgerVersion;
-}
-
-void
-LedgerManagerEditableVersion::setCurrentLedgerVersion(
-    uint32_t currentLedgerVersion)
+Application::pointer
+createTestApplication(VirtualClock& clock, Config const& cfg)
 {
-    mCurrentLedgerVersion = currentLedgerVersion;
+    auto app = Application::create(clock, cfg);
+    auto& lm = app->getLedgerManager();
+    lm.getCurrentLedgerHeader().baseFee = cfg.DESIRED_BASE_FEE;
+    testutil::setCurrentLedgerVersion(lm, cfg.LEDGER_PROTOCOL_VERSION);
+    return app;
 }
 
 void
-LedgerManagerEditableVersion::startNewLedger()
-{
-    LedgerManagerImpl::startNewLedger(1000000000000000000, mBaseFee, 100000000,
-                                      100);
-}
-
-ApplicationEditableVersion::ApplicationEditableVersion(VirtualClock& clock,
-                                                       Config const& cfg)
-    : ApplicationImpl(clock, cfg)
-{
-    mLedgerManager =
-        make_unique<LedgerManagerEditableVersion>(*this, cfg.DESIRED_BASE_FEE);
-    mHerder = Herder::create(*this); // need to recreate
-    newDB();
-}
-
-LedgerManagerEditableVersion&
-ApplicationEditableVersion::getLedgerManager()
+testutil::setCurrentLedgerVersion(LedgerManager& lm, uint32_t currentLedgerVersion)
 {
-    return static_cast<LedgerManagerEditableVersion&>(*mLedgerManager);
+    lm.getCurrentLedgerHeader().ledgerVersion = currentLedgerVersion;
```

Comment:
I really don't like the idea that this is even legal in our code to do that.... We need better encapsulation.

Reply:
Ideally this would be done by cleaning up how we use `LedgerHeaderFrame` and stop passing around just LedgerHeaders... but this is a fight for another day.